vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -36,11 +36,15 @@
 }
 
 
-def _filter_between(series: pd.Series, value: List[float]) -> pd.Series:
+def _filter_between(series: pd.Series, value: Union[List[float], List[str]]) -> pd.Series:
+    if is_datetime64_any_dtype(series):
+        value = pd.to_datetime(value)
```

Comment:
Why do we need this? Won't `value` always be a `date` since it's cast to that by pydantic?

Similarly in `_filter_isin`.

Reply:
Max has already asked the same in the https://github.com/mckinsey/vizro/pull/319#discussion_r1523452492. Let's continue the discussion there.